src/lib/orm/query.go

```diff
@@ -64,7 +64,7 @@ import (
 //			},
 //		 }
 //	}
-func QuerySetter(ctx context.Context, model interface{}, query *q.Query) (orm.QuerySeter, error) {
+func QuerySetter(ctx context.Context, model interface{}, query *q.Query, option ...string) (orm.QuerySeter, error) {
```

Comment:
1. **Error Messages**: The error messages could be more descriptive. For example, instead of saying `"<orm.QuerySetter> cannot use non-ptr model struct `%s`"`, you could say something like `"Expected a pointer to a model struct, but got a non-pointer struct: %s"`.

2. **Magic String**: The string "disableSort" is a magic string. It would be better to define it as a constant at the top of your file or package.

3. **Function Complexity**: The function is doing a lot of things - it's setting filters, sorting, and handling pagination. It might be better to split these into separate functions to reduce the complexity and improve readability.

4. **Type Checking**: The function assumes that the `model` parameter is a pointer. It might be better to add a check to ensure that it is a pointer before trying to use it.

Here's how you might refactor the function:

```go
type Option int

const (
	DisableSort Option = iota
	// Add other options here
)

func QuerySetter(ctx context.Context, model interface{}, query *q.Query, option ...Option) (orm.QuerySeter, error) {
	if err := validateModel(model); err != nil {
		return nil, err
	}

	ormer, err := FromContext(ctx)
	if err != nil {
		return nil, err
	}

	qs := ormer.QueryTable(model)
	if query == nil {
		return qs, nil
	}

	metadata := parseModel(model)
	qs = setFilters(ctx, qs, query, metadata)

	if !isSortDisabled(option) {
		qs = setSorts(qs, query, metadata)
	}

	return setPagination(qs, query), nil
}

func validateModel(model interface{}) error {
	t := reflect.TypeOf(model)
	if t.Kind() != reflect.Ptr {
		return fmt.Errorf("Expected a pointer to a model struct, but got a non-pointer struct: %s", getFullName(t.Elem()))
	}
	return nil
}

func isSortDisabled(option []Option) bool {
	for _, opt := range option {
		if opt == DisableSort {
			return true
		}
	}
	return false
}

func setPagination(qs orm.QuerySeter, query *q.Query) orm.QuerySeter {
	if query.PageSize > 0 {
		qs = qs.Limit(query.PageSize)
		if query.PageNumber > 0 {
			qs = qs.Offset(query.PageSize * (query.PageNumber - 1))
		}
	}
	return qs
}
```

This refactored version separates the concerns into smaller functions, making it easier to understand and maintain. It also uses a constant for the "disableSort" option, which makes the code more robust.

Reply:
Magic Strings are bad thing to do.

Reply:
@Vad1mo regarding the code refactor, we could raise another PR to handle it, and let's use this to focus on the beego upgrade. I'd like to have it be merged, so then we could do more testings, especially the regression parts.

Reply:
@wy65701436, Its never going to happen, I know it you know it. Another missed oppertunity, to get better.